### PR TITLE
Update logout path & refactor SessionsController

### PR DESCRIPTION
**Pull request by rrabinovitch**
This PR finalizes the login/logout set of user stories (issue #37)
• the route for logging out has been updated to be accessed via a logout link in the navbar rather than visiting a logout path. the associated route now uses a `delete` verb instead of `get`
• the navbar code has been updated to use `current_user` instead of `user` re: checking if a user is logged in and what their role is, and also now utilizes enum methods instead of string comparison to confirm a user's role
• the navbar `Log Out` link paths and verbs have been updated to align with the logout route updates noted above
• deleted duplicative root route code: used to have `application#index` AND `welcome#index`; now we just have `welcome#index`
• refactor SessionsController to DRY up code

* Once this PR is merged into the `login_logout` branch, the `login_logout` branch should have a PR submitted for merging into master

@judithpillado @ajtran303 : please note the changes re: navbar
And once this is merged into master, @ajtran303 can go ahead and update the remainder of the duplicative `current_user` code throughout the project

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -37,4 +25,32 @@ def destroy
...
+    if user.regular?
+      redirect_to '/profile'
+    elsif user.merchant_employee?
+      redirect_to '/merchant/dashboard'
+    elsif user.admin?
+      redirect_to '/admin/dashboard'
+    end
+  end
+
+  def regular_user_logged_in?
```
Comment (ajtran303) [inline app/controllers/sessions_controller.rb:41]: These Boolean conditions can be moved into the parent class as helper methods.
They can be re used in the nav bar view.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -10,10 +10,31 @@
...
   <body>
     <nav class = "topnav">
-      <%= link_to "All Merchants", "/merchants"%>
-      <%= link_to "All Items", "/items"%>
-      <%= link_to "Cart: #{cart.total_items}", "/cart" %>
-      <%= link_to "Register", "/register"%>
+      <%= link_to "Home", "/" %>
+      <%= link_to "All Merchants", "/merchants" %>
+      <%= link_to "All Items", "/items" %>
+      <% if current_user && current_user.regular? %>
```
Comment (ajtran303) [inline app/views/layouts/application.html.erb:16]: There's opportunity for code re-use here if 'regular_user_logged_in?' etc is available in this scope as a helper method in 'ApplicationController'

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -10,4 +11,19 @@ def cart
   def current_user
```
Comment (ljwhite) [inline app/controllers/application_controller.rb:11]: current_user, user, and logged_in_user are all redundant, correct?

File: app/controllers/sessions_controller.rb [hunk 4]
```diff
@@ -37,4 +25,32 @@ def destroy
...
+
+  def merchant_employee_logged_in?
+    current_user && current_user.merchant_employee?
+  end
+
+  def admin_logged_in?
+    current_user && current_user.admin?
+  end
+
+  def logged_in?
```
Comment (ljwhite) [inline app/controllers/sessions_controller.rb:53]: could you just check for @current_user?

**Comment by ljwhite** [comment]
My comments are in the code

**Pull request merged**